### PR TITLE
Remove deprecated `Compiler` module

### LINKED ISSUE #926: Remove `compilation_source` parameter
### Feature Request

| Function | Deprecated in |
| --- | --- |
| `starknet_py.common.create_compiled_contract` | 0.15.0 |
| `Contract.compute_address` | 0.14.0 |
| `Contract.compute_contract_hash` | 0.14.0 |

### PR DESCRIPTION
**Pull request by war-in**
Closes #926 

##

- [x] This PR contains breaking changes

- compilation_source
- seatch_paths
- Compiler

### CODE REVIEW

File: docs/migration_guide.rst [hunk 1]
```diff
@@ -154,7 +154,7 @@ This version deprecates several modules and fixes underlying issues with several
 0.14.0 Deprecations
```
Comment (cptartur) [inline docs/migration_guide.rst:154]: Perhaps lets already start including 0.16.0 removals in the migration guide as part of these PRs?

File: starknet_py/contract_test.py [hunk 2]
```diff
@@ -4,44 +4,6 @@
...
-
-@constructor
-func constructor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    a: felt, b: felt
-) {
-    return ();
-}
-"""
-
 SOURCE_WITH_IMPORTS = """
```
Comment (cptartur) [inline starknet_py/contract_test.py:7]: Is this still needed anywhere?

File: starknet_py/net/models/transaction_test.py [hunk 3]
```diff
@@ -110,7 +110,7 @@ def test_declare_compress_program(balance_contract):
                 nonce=23,
                 version=1,
             ),
-            1040693415700027927754435936502603457316156159253336216524207357712582758611,
+            1982828244129856379059233231530778257766878379971394245598825266532206245993,
```
Comment (drknzz) [inline starknet_py/net/models/transaction_test.py:113]: Why has this changed?

File: starknet_py/tests/e2e/mock/contracts/inner/inner.cairo
Comment (drknzz) [inline starknet_py/tests/e2e/mock/contracts/inner/inner.cairo:1]: Was this needed only for the compiler?

File: starknet_py/net/models/transaction_test.py (hunk 3 above)
Comment (war-in) [reply starknet_py/net/models/transaction_test.py:113]: I'm using different contract now

File: starknet_py/tests/e2e/mock/contracts/inner/inner.cairo
Comment (drknzz) [reply starknet_py/tests/e2e/mock/contracts/inner/inner.cairo:1]: Shouldn't `base.cairo` be removed then as well?

File: starknet_py/net/models/transaction_test.py (hunk 3 above)
Comment (drknzz) [reply starknet_py/net/models/transaction_test.py:113]: How come? I don't see any contract changes 👽

Comment (war-in) [reply starknet_py/net/models/transaction_test.py:113]: Uh, my bad. will fix that 👀

**Review by cptartur — APPROVED** [review]

**Pull request merged**